docs: improve configuration development documentation (#7707)
renovatebot/renovate: docs/development/configuration.md

## Configuration Methods

> The Renovate bot can be configured via any of these methods:

Reviewer: ```suggestion
Renovate's bot admin settings can be configured via any of these methods:
```

- Configuration file

> - Environment (variable? file? what is this about?)

Author: Can somebody answer what this is about?

> - Environment (variable? file? what is this about?)

Reviewer: ```suggestion
- Environment variables
```

> - CLI

Reviewer: ```suggestion
- CLI parameters
```

### Default Configuration

> The default configuration values can be found in [lib/config/definitions.ts](../../lib/config/definitions.ts).

Reviewer: ```suggestion
The default configuration values can be found in [lib/config/definitions.ts](../../lib/config/definitions.ts).
Options which have `"admin": true` are reserved only for bot admin configuration and cannot be configured within repository config files.
```
